addons/hr_timesheet/security/ir.model.access.csv

```diff
@@ -1,4 +1,5 @@
 id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_account_analytic_line_portal_user,analytic.account.analytic.line.timesheet.portal.user,analytic.model_account_analytic_line,base.group_portal,1,0,0,0

```

Comment:
unfortunately this permission will apply on all portal users even when the project sharing feature is not used. I wonder if we could avoid that, and only install it when the feature is added.
perhaps if this was located in a different module, only installed when you enable the feature in the settings?

Reply:
I suggest to archive/unarchive the ir.rule / ir.model.access according to the projects configuration. 

Another approach suggested by ODO would be to isolate the whole feature in a single module. However, this would require several bridge modules (timesheet, fsm, ...)